### PR TITLE
Improve lookup error handling

### PR DESCRIPTION
**Pull request by brtietz**
Return values from REopt were causing SAM to crash. Check for a null pointer earlier to avoid the crash.

To reproduce:
- Check out develop on lk and wex, https://github.com/NREL/ssc/commit/d2f76f09d4e6f346fc8ceb16f8c53b30086f38c1 on ssc, and https://github.com/NREL/SAM/commit/d498705c6a2d112613211d64dd578009b743fb27 on SAM, build
- Open a behind the meter PV-Battery case and run "Get size and dispatch from REopt"
- Observe the crash
- Check out improve_lookup_error_handling on lk, rebuild
- Open a behind the meter PV-Battery case and run "Get size and dispatch from REopt"
- Observe the improved error message and lack of crash

### CODE REVIEW

File: src/env.cpp [hunk 1]
```diff
@@ -675,6 +675,7 @@ lk::vardata_t *lk::vardata_t::index(size_t idx) const {
 }
 
 lk::vardata_t *lk::vardata_t::lookup(const lk_string &key) const {
+    if (this == 0) throw error_t(lk_tr("access violation: expected hash table, but found null")); // Check nullptr before the next line throws uncatchable errors
```
Comment (sjanzou) [inline src/env.cpp]: In XCode 14.1 this causes a warning (which is causing the Github Actions to fail):
image

Comment (sjanzou) [reply src/env.cpp]: Suggest changing to the following that was tested on MacOS 12.6 with XCode 14.1
image

**Review by sjanzou — CHANGES_REQUESTED** [review]
Remove comparing this to NULL and the results do not change using the branches in the pull request.

Also, on MacOS 12.6 with XCode 14.1, no crash occurs when running the debug build with or without the env.cpp change per attached document and latest commit.
[Lk pull issue 36 from Brian and email https.docx.zip](https://github.com/NREL/lk/files/9927061/Lk.pull.issue.36.from.Brian.and.email.https.docx.zip)

**Comment by sjanzou** [comment]
Note that the (!this) comparison does not generate a warning in XCode 14.1 but does generate a warning in the Github Actions Ubuntu build. So, either is fine. However, we must remove the -Werror from the CMakeList.txt that has been committed to this branch and pull request.

**Comment by brtietz** [comment]
Thanks, I wasn't sure we wanted to remove -Werror. Were there further changes needed after [a960030](https://github.com/NREL/lk/pull/36/commits/a9600303dc68a32d8dfa344044493443bdeacc42)? This looks good to me.

**Review by sjanzou — APPROVED** [review]
looks good

**Pull request merged**